### ISSUE TITLE
Fine-Tuning SONAR

### ISSUE THREAD

**Issue by stamatisvas** [facebookresearch/SONAR#59]
Hi,
I’m currently working with the SONAR embeddings and I'm interested in fine-tuning the model/text encoder on my custom dataset (pair of sentences) for tasks like classification and semantic similarity. Is there any documentation or guidance on how to fine-tune SONAR for these purposes?

**Comment by avidale** [comment]
Hi @stamatisvas!
SONAR encoders (as well as any other [fairseq2](https://github.com/facebookresearch/fairseq2) models) are PyTorch modules, and thus compatible with any training framework you prefer (e.g. PyTorch Lightning, Accelerate, or recently released [fairseq2 trainer](https://facebookresearch.github.io/fairseq2/stable/basics/trainer.html). Alternatively, you can write a training loop in pure pytorch, without any frameworks.

Any specific details of the fine-tuning would depend on the task, language and dataset for which you are fine-tuning the model.

**Comment by Sibgat-Ul** [comment]
Hello @avidale, i was thinking of doing some experiments (distilling to a smaller version of sonar embedding), retrain the last layers of the encoders (and maybe the decoder also) for a certain task. I mostly needed help with training, as i am not sure how do i load the encoders and only train the last layer, and use in a teacher-student framework. Can you help me with the process? 

Thanks

**Comment by mauro-belgiovine** [comment]
Hi, any guidance on this? I am also interested in fine-tuning the decoder. @avidale any chance you could provide a minimal code example in pure pytorch? Thanks!

**Comment by yhamidullah** [comment]
Hi @mauro-belgiovine If you want a pure pytorch + torch no fairseq,  we have shared ours here [mtmlt/sonar-nllb-200-1.3B](mtmlt/sonar-nllb-200-1.3B). You can train it using regular pytorch or HF trainer.

**Comment by yhamidullah** [comment]
@mauro-belgiovine It is not broken, we just put it private since Monday. Due to an ongoing paper submission, because of review policy. We  will put it back to public soon.

**Comment by mauro-belgiovine** [comment]
Thanks @avidale ! This is super helpful, I will take a look and let you know if there's any questions/comments.

**Comment by Sibgat-Ul** [comment]
thanks alot @avidale  that will help a ton.

**Comment by Sibgat-Ul** [comment]
Thanks guys, I had to ask that is there any straight forward way to get the logits for every steps easily?

Thanks.

**Comment by yhamidullah** [comment]
@Sibgat-Ul you can check here if it is what you want. https://huggingface.co/mtmlt/sonar-nllb-200-1.3B. It's now open.

**Comment by Sibgat-Ul** [comment]
yes thanks alot this will save lives lol. Seriously, thanks. @yhamidullah
